### PR TITLE
fix(tab): revert change to linking function

### PR DESCRIPTION
**Pull request by areel**
reverting commit 20047577 : return to using  the compile function rather than the link function. If tab directive using ng=transclude to inject content, the injected html can not refer to injected variables e.g ng-repeat variables. Have raised issue with more detail.

### CODE REVIEW

**Comment by wesleycho** [comment]
Can you link the issue you are referring to?

**Comment by areel** [comment]
Sorry got side lined, the background for this PR is outlined in issue #5161

File: src/tabs/tabs.js [hunk 1]
```diff
@@ -202,36 +202,38 @@ angular.module('ui.bootstrap.tabs', [])
...
+            }
+          });
 
-      scope.disabled = false;
-      if (attrs.disable) {
-        scope.$parent.$watch($parse(attrs.disable), function(value) {
-          scope.disabled = !! value;
-        });
-      }
+          scope.disabled = false;
```
Comment (icfantv) [inline src/tabs/tabs.js]: please fix the indents.  thanks.

File: src/tabs/tabs.js [hunk 2]
```diff
@@ -202,36 +202,38 @@ angular.module('ui.bootstrap.tabs', [])
...
     },
     controllerAs: 'tab',
-    link: function(scope, elm, attrs, tabsetCtrl, transclude) {
-      scope.$watch('active', function(active) {
-        if (active) {
-          tabsetCtrl.select(scope);
-        }
-      });
+    compile: function(elm, attrs, transclude) {
+        return function postLink(scope, elm, attrs, tabsetCtrl) {
```
Comment (wesleycho) [inline src/tabs/tabs.js]: This should be a two space indentation here

**Comment by wesleycho** [comment]
Going to close this as stale - in addition, I believe we will have a fix for the original issue when we enforce Angular 1.5+ support at 3.0.0. I believe this PR as is would trigger a breaking change.

**Pull request closed without merging**